### PR TITLE
DS level dialogue rework

### PR DESCRIPTION
**Pull request by m-lin1**

### CODE REVIEW

File: Scripts/Dialogue/DataStructures/2 - IntroToStacks.txt [hunk 1]
```diff
@@ -10,9 +15,9 @@ If I try to pickup an item while my inventory is full, then I can't pick it up a
...
 UseItem, however, will try to use the biggest, or rightmost, occupied index. AKA the item that was last put into the inventory.
 HIGHLIGHT_USEITEM
-This imitates the LIFO principle of Stacks, which is why we call this mode the Stack mode.
+This imitates the LIFO principle of stacks, which is why we call this mode the Stack mode.
 
-What should happen, step-by-step blah blah red key goes, then blue, but use blue or whatever
+To get to the exit, we will have to keep in mind the stack concept of LIFO. Always remember that for stacks, the last item put in is the first to be used!
 ;
 
 YEEEEEHAWWWWW pardner we made it out!
```
Comment (he-larious) [inline Scripts/Dialogue/DataStructures/2 - IntroToStacks.txt:23]: Once again are you sure we want this as the finalized dialogue 💀

File: Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt [hunk 2]
```diff
@@ -1,22 +1,25 @@
-Explain Queues & how the inventory is green + has text as to what it is
+You might have noticed that my inventory has changed modes.
+Now it is in "Queue" mode, which is indicated by it being green and having the words "Queue" above it.
+In Queue mode, my inventory acts like the data structure, queue. Let me explain more about queues.
```
Comment (he-larious) [inline Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt:3]: "In Queue mode, my inventory acts like the data structure, queue." This is real clunky as a sentence, reword it

File: Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+If the item is used up, then everything in my inventory will be shifted 1 index to the left (or in math terms (their current index - 1)). So the item at index 1 will now be in index 0, the item at index 2 will be in index 1, and so on so forth.
 
-What should happen, step-by-step blah blah blue key goes, then red, but use blue or whatever
-Now that you know how Queues work, use what you've just learned to free me from my prison of brimstone and hellfire.
+To get to the exit, we will have to keep in mind the queue concept of FIFO. Always remember that for queues, the first item put in is the first to be used!
 ;
 
-uwu
\ No newline at end of file
+YIPPPPPEEEEEEEE you did it :)
```
Comment (he-larious) [inline Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt:25]: Goofy ass Robby

File: Scripts/Dialogue/DataStructures/4 - IntroToArrays.txt [hunk 4]
```diff
@@ -1,18 +1,29 @@
-Explain Arrays or whatever & how the inventory is blue + has text as to what it is
+Oh look, my inventory has changed modes once again! This time it's in "Array" mode.
+You can tell when my inventory is in Array mode because it will always be blue and have the words "Array" above it.
+In Array mode, my inventory will act like the data structure, array.
```
Comment (he-larious) [inline Scripts/Dialogue/DataStructures/4 - IntroToArrays.txt:3]: Same as before with the queue level, this sentence is awkward

File: Scripts/Dialogue/DataStructures/4 - IntroToArrays.txt [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
-Now, lead us the way out!
+Again, this imitates how a real array would work.
+
+To move to the next room, remember to be careful about which index you put your items into. If you put one in the same index as another then you might accidentally discard a key you need.
+Also, be mindful and keep track of which key you put into which index. Make sure you don't accidentally use the wrong key on a door!
+Now get us out of here!
 ;
 
-Congrats, you made it out alive!
+RAAAAUUUUUUGHHHHHHHHHHHH LET'S GOOOOOOO WE'RE OUTTA HERE BABYYYYYYY
```
Comment (he-larious) [inline Scripts/Dialogue/DataStructures/4 - IntroToArrays.txt:28]: Robby calm tf down are you on coke rn

**Review by he-larious — COMMENTED** [review]
Oh I also forgot to mention but for the Dialogue in the 1st DS level you mention the yellow highlight for the next tile that Robby would move to. I think we should prob get rid of that statement because we're most likely also removing the highlight tile feature right? Since it doesn't contribute anything since the user can't preview where Robby is moving when they drag code blocks into the IDE

File: Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt (hunk 2 above)
Comment (m-lin1) [reply Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt:3]: is it cause I use queue twice in the sentence? Would "In this mode, my inventory acts like the data structure, queue" be better? Or "In this mode, my inventory acts like a queue data structure"?

Comment (he-larious) [reply Scripts/Dialogue/DataStructures/3 - IntroToQueues.txt:3]: "In this mode, my inventory acts like a queue data structure" I like this one

**Review by he-larious — APPROVED** [review]

**Pull request merged**